mipengine/controller/node_landscape_aggregator.py

```diff
@@ -297,27 +343,30 @@ def _gather_all_dataset_info(
                 current_labels[dataset] = datasets[dataset]
 
                 if dataset in current_datasets:
-                    current_datasets[dataset].append(node_info.id)
+                    current_datasets[dataset].append(node_id)
                 else:
-                    current_datasets[dataset] = [node_info.id]
+                    current_datasets[dataset] = [node_id]
 
             aggregated_datasets[data_model] = current_labels
             dataset_locations[data_model] = current_datasets
     return dataset_locations, aggregated_datasets
 
 
 def _get_cdes_across_nodes(
-    nodes: List[NodeInfo],
+    nodes: Dict[str, NodeInfo],
+    datasets_per_node: Dict[str, Dict[str, Dict[str, str]]],
 ) -> Dict[str, List[Tuple[str, CommonDataElements]]]:
     nodes_cdes = {}
-    for node_info in nodes:
-        node_socket_addr = _get_node_socket_addr(node_info)
-        datasets_per_data_model = _get_node_datasets_per_data_model(node_socket_addr)
+    for node_id, datasets_per_data_model in datasets_per_node.items():
+        node_socket_addr = _get_node_socket_addr(nodes[node_id])
         for data_model in datasets_per_data_model:
             cdes = _get_node_cdes(node_socket_addr, data_model)
+            if not cdes:
```

Comment:
I am not sure what are you doing here. Are you fixing the cdes problem as well inside this PR?

I would prefer to stay focused on the atomic registries in this PR as we discussed. Please remove the cdes problem fixes and any related refactoring.

Reply:
Was added by mistake thanks for reminding